js/index.js

```diff
@@ -1,100 +1,360 @@
+              click:function(){
+                $$("films_form").validate();
+                webix.confirm({
+                  title:"Form would be cleared",
+                  text:"Do you still want to continue?",
+                  type:"confirm-warning"
+                }).then(
+                  function(){
+                    $$("films_form").clear();
+                    $$("films_form").clearValidation();
+                  },
+                  function(){
+                    webix.message("Rejected");
+                  }
+                );
+              }
             },
-            {
-              view:"form",
-              width:250,
-              elements:[
-                {
-                  margin:10,//margin for all elements of rows[]
-                  rows:[
-                    {template:"EDIT FILMS", type:"section"},
-                    {view:"text", label:"Title"},
-                    {view:"text", label:"Year"},
-                    {view:"text", label:"Rating"},
-                    {view:"text", label:"Votes"},
-                  ]
-                },
-                {
-                  margin:20,
-                  cols:[
-                    {view:"button", id:"btn_add_new", value:"Add new", css:"webix_primary"},
-                    {view:"button", id:"btn_clear", value:"Clear", css:"webix_secondary"},
-                  ]
-                },{},{}
-              ]
-            }
           ]
+        },{},{}
+      ],
+      rules:{//ruls for validation
+        title:webix.rules.isNotEmpty,
+        year:function(value){
+          return (value >= 1970 && value <= 2021);
         },
-        {
-           template:"The software is provided by <a href='https://webix.com'>https://webix.com</a>. All rights reserved &#169;",
-           css:"TextAlign",
-           height:30,
-           width:550
+        rating:function(value){
+          //console.log(value.replace(",", ".")); //for correct validation of float type
+          return (value.replace(",", ".") > 0 && value.replace(",", ".") <= 10);
+        },
+        votes:function(value){
+          if(value == 0 || (value.replace(",", ".") > 0 && value.replace(",", ".") < 100000)){
+            return value;
+          }else{return false}
         }
+      }
+      /*onValidationError:function(key, data){
+        webix.message.position = "bottom";
+        webix.message({text:key.toUpperCase()+"  field is incorrect", type:"error"});
+      }*/
+    };
+  var list = {
+    type:"clean",
+    rows:[
+      {
+        height: 35,
+        view:"toolbar",
+        elements:[
+          {
+            view:"text",
+            id:"list_input",
+            on:{
+              onTimedKeyPress:function(){
+                var value = this.getValue().toLowerCase();
+                $$("list").filter(function(obj){
+                    return obj.name.toLowerCase().indexOf(value)==0;
```

Comment:
it is more user-friendly to search by any position in a string, not the first one

Reply:
Ok. I have used: "return obj.name.toLowerCase().indexOf(value) !== -1;" instead of "return obj.name.toLowerCase().indexOf(value)==0;" and have got "more user-friendly" resolt.